### DISCUSSION TITLE
Component Rendered Twice with generateStaticParams

### DISCUSSION THREAD

**Discussion by lsvargas** [vercel/next.js#43158] [Help]
Hello, my component is rendered twice in production environment or develop (without stricmode). The code looks like this:

```javascript
import { expensiveFunction } from '../../../lib/expensiveFunction'

export const revalidate = 60

export async function generateStaticParams () {
  return [{ id: '1' }, { id: '2' }]
}

async function Component ({ params }) {
  const { id } = params
  const data = await expensiveFunction(id)

  return (
    <div>
       <h1>Component</h1>
    </div>
  )
}

export default Component
```
The component above renders twice, but if I remove the `generateStaticParams` It only renders once. 
This happens does not have to do with my `expensiveFunction` tho I put it there because each call to that function takes 5s and I am getting timeout from the serverless functions because I am going over the 10s allowed.

**Comment by icyJoseph** [comment]
Yeah I was able to reproduce this. Wild guess here but perhaps the props passed are recreated somehow... since there's no server side dev tools, it is hard to find out what's going on :) 

I made this naive thing:

```tsx
export const revalidate = 60;

export async function generateStaticParams() {
  return [{ id: "1" }, { id: "2" }];
}

let map = new WeakSet();

async function Component({ params, searchParams }: { params: { id: string } }) {
  const { id } = params;

  console.log("seen params?", map.has(params));
  console.log("seen search params?", map.has(searchParams));

  map.add(params);
  map.add(searchParams);

  return (
    <div>
      <h1>Component</h1>
    </div>
  );
}

export default Component;

```

And I see that params is new on the second pass, while search params is stable. Of course this checks relies on the double rendering actually happening.
